### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ChungYiTsai**

### CODE REVIEW

File: app/infrastructure/database/orms/target_orm.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    class TargetOrm < Sequel::Model(:targets)
+      one_to_many :articles,
+                  class: :'PortfolioAdvisor::Database::ArticleOrm',
+                  key: :company_id
+
+      plugin :timestamps, update_on_create: true
+
+      # def self.find_or_create(target_info)
+      #   first(company_name: target_info[:company_name]) || create(target_info)
+      # end
```
Comment (tienyulin) [inline app/infrastructure/database/orms/target_orm.rb:17]: Please remove these comment codes.

File: app/infrastructure/gateways/google_news_api.rb [hunk 2]
```diff
@@ -25,21 +25,20 @@ def initialize(token)
...
-
-          path = "q=#{topic}&from=#{from}&to=#{to}&pageSize=#{result_num}"
-          # path = "q=#{topic}&from=2021-10-10&to=2021-10-20&pageSize=#{result_num}"
-          #  puts path
+          path = "q=#{topic}&from=2021-10-10&to=2021-10-20&pageSize=#{result_num}"
+          # the comment below is for vcr
+          # today = Date.today
+          # to = today.strftime('%Y-%m-%d')
+          # from = (today - 10).strftime('%Y-%m-%d')
+          # path = "q=#{topic}&from=#{from}&to=#{to}&pageSize=#{result_num}"
```
Comment (tienyulin) [inline app/infrastructure/gateways/google_news_api.rb:33]: I know you want to keep it for next time if VCR changes the result. However, pushing these comment codes will make it untidy and confuse others.
In the industry, people will leave and new members will join this team, so it's really hard for new members to understand what are these comment codes for.
I would suggest you post this into [Wiki](https://github.com/SOA-nuts/GoogleNewsAPI/wiki), Wiki is the place for elaborating more about your code and how to use it for your team members.
Thus, you can copy these codes then paste them while you need them. That's the way the industry does, always make the code readable, understandable, and no useless code and comment.

File: app/models/entities/article.rb [hunk 3]
```diff
@@ -12,7 +13,12 @@ class Article < Dry::Struct
 
       attribute :title,             Strict::String
       attribute :url,               Strict::String
-      attribute :published_at,      Publish
+      attribute :published_at,      Strict::DateTime
+      # attribute :published_date, Strict::String
+      # attribute :published_time, Strict::String
```
Comment (tienyulin) [inline app/models/entities/article.rb:18]: Please remove these commented codes.

File: app/models/mappers/article_mapper.rb [hunk 4]
```diff
@@ -27,35 +23,38 @@ def self.build_entity(article)
...
           @data = data
-          @publish_mapper = PublishMapper.new(@data['publishedAt'])
+          @publish_at = DateTime.strptime(@data['publishedAt'], '%Y-%m-%dT%H:%M:%S%z')
         end
 
         def build_entity
           PortfolioAdvisor::Entity::Article.new(
             url: url,
+            # published_date: published_date,
+            # published_time: published_time,
```
Comment (tienyulin) [inline app/models/mappers/article_mapper.rb:33]: Please remove these commented codes.

File: app/models/mappers/article_mapper.rb [hunk 5]
```diff
@@ -27,35 +23,38 @@ def self.build_entity(article)
...
         end
 
+        # def published_time
+        #   @publish_at.strftime('%H:%M:%S')
+        # end
+
+        # def published_date
+        #   # @publish_at.strftime('%Y-%m-%d')
+        # end
+
```
Comment (tienyulin) [inline app/models/mappers/article_mapper.rb:52]: Please remove these commented codes.

**Review by tienyulin — COMMENTED** [review]
Issue #16, Reviewed.

File: app/infrastructure/gateways/google_news_api.rb (hunk 2 above)
Comment (ChungYiTsai) [reply app/infrastructure/gateways/google_news_api.rb:33]: Got it.
Thanks for advice!

**Pull request merged**